src/main/java/bio/terra/cli/service/WorkspaceManager.java

```diff
@@ -324,156 +476,181 @@ private static boolean isValidEnvironmentVariableName(String name) {
    *
-   * @param referenceName name of reference to lookup
-   * @return the data reference object
-   * @throws UserActionableException if the resource is not a data reference (e.g. VM)
+   * @param name name of the resource. this is unique across all resources in the workspace
+   * @return the resource description object that was deleted
    */
-  public CloudResource getDataReference(String referenceName) {
-    workspaceContext.requireCurrentWorkspace();
-    // TODO: change this method to call WSM data reference endpoints once they're ready
-    CloudResource dataReference = workspaceContext.getCloudResource(referenceName);
-    if (dataReference == null) {
-      throw new UserActionableException(referenceName + " not found.");
-    }
-    if (!dataReference.type.isDataReference) {
-      throw new UserActionableException(dataReference + " is not a data reference.");
-    }
-    return dataReference;
+  public ResourceDescription deleteControlledGcsBucket(String name) {
+    return deleteResource(
+        name,
+        (resourceId) -> {
+          new WorkspaceManagerService(globalContext.server, globalContext.requireCurrentTerraUser())
+              .deleteControlledGcsBucket(workspaceContext.getWorkspaceId(), resourceId);
+        });
   }
 
   /**
-   * Add a new data reference in the workspace.
+   * Delete a Big Query dataset controlled resource in the workspace. Also updates the cached list
+   * of resources.
    *
-   * <p>This method checks that the data reference exists using the user's credentials. The
-   * reference does not have to be within the workspace project.
+   * @param name name of the resource. this is unique across all resources in the workspace
+   * @return the resource description object that was deleted
+   */
+  public ResourceDescription deleteControlledBigQueryDataset(String name) {
+    return deleteResource(
+        name,
+        (resourceId) -> {
+          new WorkspaceManagerService(globalContext.server, globalContext.requireCurrentTerraUser())
+              .deleteControlledBigQueryDataset(workspaceContext.getWorkspaceId(), resourceId);
+        });
+  }
+
+  /**
+   * Delete a resource in the workspace. Also updates the cached list of resources.
    *
-   * @param referenceType type of reference to add
-   * @param referenceName name of reference to add
-   * @param cloudId unique identifier for this resource in the cloud (e.g. bucket uri, bq dataset
-   *     id)
-   * @return the data reference that was added
+   * @param resourceName name of resource to delete
+   * @param deleteResource function pointer to execute the delete request for a particular
+   *     stewardship + resource type combination
+   * @return the resource description object that was deleted
    */
-  public CloudResource addDataReference(
-      CloudResource.Type referenceType, String referenceName, String cloudId) {
+  private ResourceDescription deleteResource(String resourceName, Consumer<UUID> deleteResource) {
```

Comment:
ditto comment on createResource

Reply:
Yeah, I'm on the fence about whether these are worth it, too. It was just a code de-duplication effort, basically every `delete...` method needs these lines before:
```
    workspaceContext.requireCurrentWorkspace();
    // get the summary object
    ResourceDescription resourceToDelete = workspaceContext.getResource(resourceName);
```
and these lines after:
```
    updateResourcesCache(); // persist the cloud resource locally
    return resourceToDelete; // return just the summary object
```
Similar for `create`. It's not a lot of code, I was just thinking that the next time we add a new resource, I won't have to remember to copy that. Thoughts?